src/state_machine/coordinator/fire.rs

```diff
@@ -1110,9 +1110,13 @@ impl<Aggregator: AggregatorTrait> Coordinator<Aggregator> {
             .cloned()
             .flat_map(|pn| pn.nonces)
             .collect::<Vec<PublicNonce>>();
-        let (_, R) = compute::intermediate(&self.message, &party_ids, &nonces);
+        if let Some(group_key) = self.aggregate_public_key {
```

Comment:
It reads better and we do less work if we move this to the beginning of the function and do
```rust
let Some(group_key) = self.aggregate_public_key else {
    return Err(Error::MissingAggregatePublicKey);
}
```
same thing for the other `compute_aggregate_nonce` implementation.

Reply:
I've generally come to like the `let () else` paradigm, though in this case where we're at the tail of the `fn` I can go either way.

Fixed here and in `frost::Coordinator`.